### DISCUSSION TITLE
Rule proposal: template/no-variable-assignment

### DISCUSSION THREAD

**Discussion by mgechev** [angular-eslint/angular-eslint#819] [New Rule Proposals]
This rule should report if an Angular template contains property assignments to template variables.

Currently, there's no equivalent of such a rule in codelyzer, but is something we can consider adding to the CLI.

**Comment by arturovt** [comment]
Right now this is built-in into the Ivy compiler since it throws exceptions when the template contains assignments to template variables:
`
Cannot use variable 'VARIABLE' as the left-hand side of an assignment expression. Template variables are read-only.
`

**Comment by mgechev** [comment]
I believe that's not targeting template variables, but any field declared in the component's constructor.
